qiskit/primitives/base_estimator.py

```diff
@@ -421,10 +421,10 @@ def __call__(
 
     def run(
         self,
-        circuits: Sequence[QuantumCircuit],
-        observables: Sequence[BaseOperator | PauliSumOp],
-        parameter_values: Sequence[Sequence[float]] | None = None,
-        parameters: Sequence[Sequence[Parameter]] | None = None,
+        circuits: QuantumCircuit | Sequence[QuantumCircuit],
+        observables: BaseOperator | PauliSumOp | Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[float] | Sequence[Sequence[float]] | None = None,
+        parameters: Sequence[Parameter] | Sequence[Sequence[Parameter]] | None = None,
```

Comment:
I might be out-of-date, because I don't track the primitives implementation, but I'm pretty sure it was originally a specific design decision to exclude this - the comment chain is here: https://github.com/Qiskit/qiskit-terra/pull/7723#discussion_r818516373.

If there's been more discussion with @levbishop, @ajavadia (and anyone else) that motivates overriding that, then please just ignore me.

Reply:
FYI: I have a PR #8684 to remove `parameters` and wait for response of Lev and Ali.

Reply:
> I might be out-of-date, because I don't track the primitives implementation, but I'm pretty sure it was originally a specific design decision to exclude this - the comment chain is here: [#7723 (comment)](https://github.com/Qiskit/qiskit-terra/pull/7723#discussion_r818516373).

I created this PR because I found it very tedious to always have to specify`[[]]`. And at least 5 other people shared the same sentiment on slack 😁. I don't disagree it could lead to tricky inference issue, but I think the benefit outweighs the issue - i.e. more people want to avoid putting a single value in a list than those who want to pass an empty list.

Reply:
I think @levbishop's primary objection was to automatic broadcasting behavior. I don't think that extends to supporting single objects outside of lists where there is no confusion about order.